wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+        for the text, positioning them centered vertically.
+
+        :param `dc`: device context to use.
+        :type `dc`: `wx.DC`
+        """
+
+        # Get the actual client size of ourselves.
+        width, height = self.GetClientSize()
+
+        if not width or not height:
+            # Nothing to do, we still don't have dimensions!
+            return
+
+        # Initialize the wx.BufferedPaintDC, assigning a background
+        # colour and a foreground colour (to draw the text).
+        backColour = self.GetBackgroundColour()
+        backBrush = wx.Brush(backColour, wx.SOLID)
+        dc.SetBackground(backBrush)
+        dc.Clear()
+
+        if self.IsEnabled():
+            dc.SetTextForeground(self.GetForegroundColour())
+        else:
+            dc.SetTextForeground(self.SYS_COLOUR_GRAYTEXT)
+
+        dc.SetFont(self.GetFont())
+
+        # Get the text label for the checkbox, the associated check bitmap
+        # and the spacing between the check bitmap and the text.
+        label = self.GetLabel()
+        bitmap = self.GetBitmap()
+        spacing = self.GetSpacing()
+
+        # Measure the text extent and get the check bitmap dimensions.
+        textWidth, textHeight = dc.GetTextExtent(label)
+        bitmapWidth, bitmapHeight = bitmap.GetWidth(), bitmap.GetHeight()
+
+        # Position the bitmap centered vertically.
+        bitmapXpos = 0
+        bitmapYpos = (height - bitmapHeight) // 2
+
+        # Position the text centered vertically.
+        textXpos = bitmapWidth + spacing
+        textYpos = (height - textHeight) // 2
+
+        # Draw the bitmap on the DC.
+        try:
+            dc.DrawBitmap(bitmap, bitmapXpos, bitmapYpos, True)
+        except Exception as exc:  # bitmap might be image and need converted. Ex: if disabled.
+            dc.DrawBitmap(bitmap.ConvertToBitmap(), bitmapXpos, bitmapYpos, True)
+
+        # Draw the text
+        dc.DrawText(label, textXpos, textYpos)
+
+        # Let's see if we have keyboard focus and, if this is the case,
+        # we draw a dotted rectangle around the text (Windows behavior,
+        # I don't know on other platforms...).
+        if self.HasFocus():
+            # Yes, we are focused! So, now, use a transparent brush with
+            # a dotted black pen to draw a rectangle around the text.
+            dc.SetBrush(wx.TRANSPARENT_BRUSH)
+            dc.SetPen(self._focusIndPen)
+            dc.DrawRectangle(textXpos, textYpos, textWidth, textHeight)
+
+    def OnEraseBackground(self, event):
+        """
+        Handles the ``wx.EVT_ERASE_BACKGROUND`` event for :class:`GenCheckBox`.
+
+        :param `event`: A `wx.EraseEvent` to be processed.
+        :type `event`: `wx.EraseEvent`
+        """
+
+        # This is intentionally empty, because we are using the combination
+        # of wx.BufferedPaintDC + an empty OnEraseBackground event to
+        # reduce flicker.
+        pass
```

Comment:
Since the paint event draws the whole widget, then you can use `SetBackgroundStyle(wx.BACKGROUND_STYLE_PAINT)` and then implementing an erase-background handler is not neccessary.

Reply:
Removed, but added in a OnSize refresh to prevent smearing.